src/main/java/com/gmail/goosius/siegewar/utils/SiegeWarBattleSessionUtil.java

```diff
@@ -139,6 +132,40 @@ public void run() {
 		}
 	}
 
+	/** 
+	 * Determines how many points are awarded to the winner of a Battle Session.
+	 *
+	 * Affected by the areBattlePointsWinnerTakesAll config setting.
+	 * 
+	 * @param siege the Siege to gather BattlePoints from.
+	 * @return points awarded to the winner.
+	 */
+	public static int calcPointsForBattleSession(Siege siege) {
+		int battlePointsOfWinner;
+		
+		// Attackers won the session.
+		if (siege.getAttackerBattlePoints() > siege.getDefenderBattlePoints()) {
+			
+			if (SiegeWarSettings.areBattlePointsWinnerTakesAll())
+				battlePointsOfWinner = siege.getAttackerBattlePoints();
+			else 
+				battlePointsOfWinner = siege.getAttackerBattlePoints() - siege.getDefenderBattlePoints();
+
+		// Defenders won the session.
+		} else if (siege.getAttackerBattlePoints() < siege.getDefenderBattlePoints()) {
+
+			if (SiegeWarSettings.areBattlePointsWinnerTakesAll())
+				battlePointsOfWinner = -siege.getDefenderBattlePoints();
+			else 
+				battlePointsOfWinner = -(siege.getDefenderBattlePoints() - siege.getAttackerBattlePoints());
```

Comment:
:construction: 
- The above calculation should be 
```
siegePointsAdjustment = siege.getAttackerBattlePoints() - siege.getDefenderBattlePoints();
```
- Or to put it another way, as long as winnerTakesAll is configured to false, the above is the formula to use, regardless of who won.

Reply:
Hmmm, yes that is correct, I will make this simpler.